tests/integration/test_rescale_loki_push_api.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""This test module tests how relation data changes with rescaling.
+
+Feature: When the Loki charm is rescaled, all the related charms over loki_push_api need to have
+up-to-date information of current (remaining) Loki units.
+
+Scenario:
+1. Deploy the charm with one unit.
+2. Scale up from one to three units.
+3. Scale down to zero units.
+"""
+
```

Comment:
I have mixed feelings about this integration test. The `show_unit` helper is *great*. Relying on it for workflow doesn't give any assurances that the data written "makes it" to anything useful, though. @balbirthomas did a great job with `test_alert_rules_fire` which could probably mostly be re-used here (or just add/change that test so we don't have more instances of the "test this with a single unit, then have a whole different build/model/etc for multiple" pattern).

When we scale up, fire the action and make sure alerts flag. That's a reasonable check for "logs made it to the endpoint and were processed." Otherwise, we've all seen enough instances of "this made it into juju relation data but the workload lost it somewhere in the mix"

Reply:
I just deleted the entire file (the test suite is long enough) and added verification in `test_alert_rules_fire` with an additional testcase which actually scales the units up and makes sure they fire.

Reply:
We got rid of `test_rescale_charm.py` since it was not a very focused test and `test_rescale_loki_push_api.py` was added to replace it.... 

As far as I understand, `test_alert_rules_fire` tests are not checking for Loki endpoints in relation data..... or am I missing something?